Documentation/upgrade.md

```diff
@@ -157,42 +157,54 @@ The agents are deployed and managed by a Kubernetes daemonset. So in order to up
 
 The following command updates the image of the agent daemonset:
 ```bash
-kubectl -n rook-system set image daemonset/rook-agent rook-agent=rook/rook:v0.5.1
+kubectl -n rook-system set image daemonset/rook-agent rook-agent=rook/rook:master
 ```
 
-Once the daemonset is updated, we can begin deleting each agent pod **one at a time** and verifying a new one comes up to replace it that is running the new version.
+Once the command is executed, Kubernetes will begin the flow of updating each of the agent pods.
+
+#### Upgrading from 0.6.x or lower
+If upgrading for a Rook version of 0.6.x or lower, the daemonset pods will need to be manually deleted in order to be replaced with the new version.
+Each agent pod will have to be deleted **one at a time** and verifying a new one comes up to replace it that is running the new version.
 
 The following is an example of deleting just one of the agent pods (note that the names of your agent pods will be different):
 ```bash
 kubectl -n rook-system delete pod rook-agent-56m61
 ```
 
+#### Agent Health Verification
 After all the agent pods have been updated, verify that they are `Running`.
 ```bash
 kubectl -n rook-system get pod -l app=rook-agent
 ```
 
 ### API
 Similar to the operator, the Rook API pod is managed by a Kubernetes deployment.
-However, this time we will edit the deployment directly, because there is another field besides just the image version that needs to be updated in the Rook API deployment.
+However, this time we will edit the deployment directly, because there are other fields that need to be added besides updating the image version in the Rook API deployment.
 Begin editing the deployment with:
 ```bash
 kubectl -n rook edit deployment rook-api
 ```
 
-Note there are 2 fields that need to be updated.
-These are the two relevant snippets you should update in your editor to use the new version of `rook/rook:v0.5.1` as shown:
-```
-        - name: ROOK_VERSION_TAG
-          value: v0.5.1
-```
+First update the image version to the new version of `rook/rook:master` as shown in the following snippet:
```

Comment:
the api pods are also stateless. Let's just tell them to delete its deployment before upgrade as well. Much simpler than trying to modify the pod spec

Reply:
Will the operator deploy a delta of the cluster? IE if some of the component already exist but some dont, would the operator bring up the delta?

Reply:
when the operator starts, it will look at each first-level component and make sure it exists. The mons, osds, api, agent, rgw, etc will all be independent. But if a deployment exists, the operator does *not* go any deeper to see if the pod spec is up-to-date.